e2e/nomostest/config_sync.go

```diff
@@ -816,6 +817,128 @@ func RepoSyncObjectV1Beta1(nn types.NamespacedName, repoURL string, sourceFormat
 	return rs
 }
 
+// UpdateRootSyncSpec updates the RootSync object on the cluster with the provided
+// spec.
+func (nt *NT) UpdateRootSyncSpec(rs *v1beta1.RootSync) error {
+	var clusterRootSync v1beta1.RootSync
+	if err := nt.KubeClient.Get(rs.Name, rs.Namespace, &clusterRootSync); err != nil {
+		return err
+	}
+	clusterRootSync.Spec = rs.Spec
+	return nt.KubeClient.Update(&clusterRootSync)
+}
+
+// UpdateRepoSyncSpec updates the RepoSync object on the cluster with the provided
+// spec.
+func (nt *NT) UpdateRepoSyncSpec(rs *v1beta1.RepoSync) error {
+	var clusterRepoSync v1beta1.RepoSync
+	if err := nt.KubeClient.Get(rs.Name, rs.Namespace, &clusterRepoSync); err != nil {
+		return err
+	}
+	clusterRepoSync.Spec = rs.Spec
+	return nt.KubeClient.Update(&clusterRepoSync)
+}
+
+// RootSyncObjectOCI returns a RootSync object that syncs the provided OCIImage.
+func (nt *NT) RootSyncObjectOCI(name string, image *registryproviders.OCIImage) *v1beta1.RootSync {
+	rs := RootSyncObjectV1Beta1FromRootRepo(nt, name)
+	rs.Spec.SourceType = string(v1beta1.OciSource)
+	rs.Spec.Oci = &v1beta1.Oci{
+		Image: image.FloatingBranchTag(),
+		Auth:  configsync.AuthNone,
+	}
+	switch *e2e.OCIProvider {
+	case e2e.Local:
+		// Local provider requires CA cert because host is not well known
+		rs.Spec.Oci.CACertSecretRef = &v1beta1.SecretReference{
+			Name: PublicCertSecretName(RegistrySyncSource),
+		}
+	case e2e.ArtifactRegistry:
+		// AR provider requires auth because the registry is private
+		rs.Spec.Oci.Auth = configsync.AuthGCPServiceAccount
+		rs.Spec.Oci.GCPServiceAccountEmail = registryproviders.ArtifactRegistryReaderEmail()
+	}
+	return rs
+}
+
+// RepoSyncObjectOCI returns a RepoSync object that syncs the provided OCIImage.
+func (nt *NT) RepoSyncObjectOCI(nn types.NamespacedName, image *registryproviders.OCIImage) *v1beta1.RepoSync {
+	rs := RepoSyncObjectV1Beta1FromNonRootRepo(nt, nn)
+	rs.Spec.SourceType = string(v1beta1.OciSource)
+	rs.Spec.Oci = &v1beta1.Oci{
+		Image: image.FloatingBranchTag(),
+		Auth:  configsync.AuthNone,
+	}
+	switch *e2e.OCIProvider {
+	case e2e.Local:
+		// Local provider requires CA cert because host is not well known
+		rs.Spec.Oci.CACertSecretRef = &v1beta1.SecretReference{
+			Name: PublicCertSecretName(RegistrySyncSource),
+		}
+	case e2e.ArtifactRegistry:
+		// AR provider requires auth because the registry is private
+		rs.Spec.Oci.Auth = configsync.AuthGCPServiceAccount
+		rs.Spec.Oci.GCPServiceAccountEmail = registryproviders.ArtifactRegistryReaderEmail()
+	}
```

Comment:
since you have the nt, you can fail the test in the default case where the provider is invalid.

Reply:
The providers should already be validated at this point, but no harm in adding it here. Done